### DISCUSSION TITLE
[Enhancement]: Deploy different versions of Strimzi operators in a single k8s cluster, allowing the cluster to deploy different versions of Kafka clusters

### DISCUSSION THREAD

**Discussion by junlintianxiazhifulinzhongguo** [strimzi/strimzi-kafka-operator#10087] [General]
### Related problem

Deploying the Strimzi operator in k8s has compatibility with relevant versions. If I deploy the 0.40.0 version of Strimzi operator in k8s v1.23.17, then I can only deploy the 3.6.0, 3.6.1, and 3.7.0 versions of Kafka. If I have a requirement at this time to deploy versions 3.0. x to 3.5. x of Kafka, the previously deployed versions 3.6.0, 3.6.1, and 3.7.0 of Kafka will still be retained. In this case, I need to deploy Strimzi operators below 0.40.0 in the cluster, which will cause CRD conflicts.

### Suggested solution

Is there any way to deploy all Strimzi operators compatible with k8s versions in independent spaces, specifying namespaces that can be managed and dynamically modified? Different versions of Strimzi operators can be deployed in a k8s cluster, and the namespaces that can be used by CRDs can be limited. Different CRD versions should be isolated from each other without causing CRD conflicts, so that different versions of Kafka can be deployed in a k8s cluster.

**Comment by scholzj** [comment]
This is not something what is supported by Kubernetes.
